### PR TITLE
Moving AtClient to a separate library.

### PR DESCRIPTION
**Pull request by slaff**
Adding a sample (related to #1527).

### CODE REVIEW

**Comment by slaff** [comment]
@mikee47 I am thinking that maybe AtClient can be completely rewritten to use your [IO Control](https://sming.readthedocs.io/en/latest/_inc/Sming/Libraries/IOControl/index.html) library.

File: Sming/Libraries/AtClient/README.rst [hunk 1]
```diff
@@ -0,0 +1,39 @@
+At Client
+=========
+
+.. highlight:: c++
+
+Introduction
+------------
+
+AT commands were initially instructions used to control a modem. AT is the abbreviation of ATtention. Every command line starts with "AT" or "at".
```
Comment (mikee47) [inline Sming/Libraries/AtClient/README.rst:9]: Good background article https://en.wikipedia.org/wiki/Hayes_command_set

**Comment by mikee47** [comment]
> @mikee47 I am thinking that maybe AtClient can be completely rewritten to use your [IO Control](https://sming.readthedocs.io/en/latest/_inc/Sming/Libraries/IOControl/index.html) library.

@slaff Probably a good fit as it's a bit like modbus with a structure command/response. Modems are particularly interesting since they all do a similar job, use AT commands but don't always implement them consistently! This might be accommodated using a basic AT IOControl::device which can be configured with fixed (i.e. flash-based) mappings for various device types.

**Review by mikee47 — APPROVED** [review]
Looks fine. I am interested in adding GSM modem support to the IOControl library, though I'm in the middle of a solar power installation ATM. Feel free to have a go!

**Pull request merged**